<a href="https://colab.research.google.com/github/mmamoun1/Unit3FinalProject/blob/main/Unit3Final_MostafaMamoun.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Unit 3 Final

https://healthdata.gov/dataset/COVID-19-Reported-Patient-Impact-and-Hospital-Capa/6xf2-c3ie

Data Description: This data is a collection of "COVID-19 Reported Patient Impact and Hospital Capacity by State", with variables describing how COVID 19 is impacting thousands of hospitals across the United States.

In [ ]:
# First Step: I will import pandas to use the dataframe functionality
import pandas as pd
# Now, I can import the dataset into a dataframe named df.
df = pd.read_csv('https://healthdata.gov/resource/6xf2-c3ie.csv')
# Now, I can check the shape of the dataframe to see if it has the expected 72.6k rows of data'
df.shape

(54, 135)

If the shape is (54, 135), then the dataset has been read in correctly and can now be manipulated.

In [ ]:
## This dataset has a lot of valuable information about the number of hospitals in each state that are being affected by COVID-19, but
# the amount of information can make it hard to find the most pressing issues. So, I will subset the data to create a new dataframe with
# the 10 states with the most anticipated critical staffing shortages within the week. This way, focus can be put on those states who require the most assistance,
# and future analysis can also be done on that state's hospital information to find out why these shortages occurred and predict them occurring again.

## The first step to do this is to sort by anticipated hospital shortages.
##Using the sort_values function, we can sort by the necessary column and set it to descending
df = df.sort_values(by=['critical_staffing_shortage_anticipated_within_week_yes'], ascending=False)

# Then, we can create a new dataframe for the first 10 values in the sorted dataframe using the head() function
top_ten_df = df.head(10)
# Now, we can print out the new dataframe and observe the results
top_ten_df






state  critical_staffing_shortage_today_yes  \
45    MI                                    28   
3     CA                                    10   
33    PA                                     1   
8     MO                                     1   
17    MD                                    12   
28    TN                                     3   
49    WI                                    14   
1     NM                                     0   
22    AR                                     1   
7     OK                                     0   

    critical_staffing_shortage_today_no  \
45                                  107   
3                                   114   
33                                    9   
8                                    19   
17                                    4   
28                                   74   
49                                   72   
1                                     2   
22                                    7   
7                                     1   

    critical_staffing_shortage_today_not_reported  \
45                                             21   
3                                             223   
33                                            178   
8                                              94   
17                                             34   
28                                             36   
49                                             52   
1                                              44   
22                                             82   
7                                             126   

    critical_staffing_shortage_anticipated_within_week_yes  \
45                                                 31        
3                                                  24        
33                                                 19        
8                                                  19        
17                                                 14        
28                                                 13        
49                                                 12        
1                                                  12        
22                                                 12        
7                                                  12        

    critical_staffing_shortage_anticipated_within_week_no  \
45                                                121       
3                                                 256       
33                                                163       
8                                                  59       
17                                                 17       
28                                                 77       
49                                                 88       
1                                                  31       
22                                                 75       
7                                                  90       

    critical_staffing_shortage_anticipated_within_week_not_reported  \
45                                                  4                 
3                                                  67                 
33                                                  6                 
8                                                  36                 
17                                                 19                 
28                                                 23                 
49                                                 38                 
1                                                   3                 
22                                                  3                 
7                                                  25                 

    hospital_onset_covid  hospital_onset_covid_coverage  inpatient_beds  ...  \
45                   1.0                             26           21290  ...   
3                   18.0                            170           60313  ...   
33               